--- php5.orig/ext/curl/interface.c
+++ php5/ext/curl/interface.c
@@ -2183,7 +2183,7 @@ string_copy:
 
 			convert_to_string_ex(zvalue);
 
-			if (!Z_STRLEN_PP(zvalue) || php_check_open_basedir(Z_STRVAL_PP(zvalue) TSRMLS_CC)) {
+			if (Z_STRLEN_PP(zvalue) && php_check_open_basedir(Z_STRVAL_PP(zvalue) TSRMLS_CC)) {
 				RETVAL_FALSE;
 				return 1;
 			}
--- php5.orig/ext/curl/tests/bug61948.phpt
+++ php5/ext/curl/tests/bug61948.phpt
@@ -13,7 +13,7 @@ open_basedir="/tmp"
   curl_close($ch);
 ?>
 --EXPECTF--
-bool(false)
+bool(true)
 bool(true)
 
 Warning: curl_setopt(): open_basedir restriction in effect. File(/xxx/bar) is not within the allowed path(s): (/tmp) in %sbug61948.php on line %d
