commit d08b4dbf23febd3f305a2682b03ab9c70f11ac60
Author: Michael Wallner <mike@php.net>
Date:   Thu Apr 3 10:40:06 2014 +0200

    Fix Bug #66736 	fpassthru broken

--- php5.orig/main/output.c
+++ php5/main/output.c
@@ -234,6 +234,13 @@ PHPAPI int php_output_get_status(TSRMLS_
  * Unbuffered write */
 PHPAPI int php_output_write_unbuffered(const char *str, size_t len TSRMLS_DC)
 {
+#if PHP_DEBUG
+	if (len > UINT_MAX) {
+		php_error(E_WARNING, "Attempt to output more than UINT_MAX bytes at once; "
+				"output will be truncated %lu => %lu",
+				(unsigned long) len, (unsigned long) (len % UINT_MAX));
+	}
+#endif
 	if (OG(flags) & PHP_OUTPUT_DISABLED) {
 		return 0;
 	}
@@ -248,6 +255,13 @@ PHPAPI int php_output_write_unbuffered(c
  * Buffered write */
 PHPAPI int php_output_write(const char *str, size_t len TSRMLS_DC)
 {
+#if PHP_DEBUG
+	if (len > UINT_MAX) {
+		php_error(E_WARNING, "Attempt to output more than UINT_MAX bytes at once; "
+				"output will be truncated %lu => %lu",
+				(unsigned long) len, (unsigned long) (len % UINT_MAX));
+	}
+#endif
 	if (OG(flags) & PHP_OUTPUT_DISABLED) {
 		return 0;
 	}
--- php5.orig/main/streams/streams.c
+++ php5/main/streams/streams.c
@@ -1404,11 +1404,16 @@ PHPAPI size_t _php_stream_passthru(php_s
 		p = php_stream_mmap_range(stream, php_stream_tell(stream), PHP_STREAM_MMAP_ALL, PHP_STREAM_MAP_MODE_SHARED_READONLY, &mapped);
 
 		if (p) {
-			PHPWRITE(p, mapped);
+			do {
+				/* output functions return int, so pass in int max */
+				if (0 < (b = PHPWRITE(p, MIN(mapped - bcount, INT_MAX)))) {
+					bcount += b;
+				}
+			} while (b > 0 && mapped > bcount);
 
 			php_stream_mmap_unmap_ex(stream, mapped);
 
-			return mapped;
+			return bcount;
 		}
 	}
 
