--- php5.orig/ext/intl/collator/collator_class.c
+++ php5/ext/intl/collator/collator_class.c
@@ -67,6 +67,7 @@ zend_object_value Collator_object_create
 	intern = ecalloc( 1, sizeof(Collator_object) );
 	intl_error_init( COLLATOR_ERROR_P( intern ) TSRMLS_CC );
 	zend_object_std_init( &intern->zo, ce TSRMLS_CC );
+	object_properties_init(&intern->zo, ce);
 
 	retval.handle = zend_objects_store_put(
 		intern,
--- php5.orig/ext/intl/dateformat/dateformat_class.c
+++ php5/ext/intl/dateformat/dateformat_class.c
@@ -63,6 +63,7 @@ zend_object_value IntlDateFormatter_obje
 	intern = ecalloc( 1, sizeof(IntlDateFormatter_object) );
 	dateformat_data_init( &intern->datef_data TSRMLS_CC );
 	zend_object_std_init( &intern->zo, ce TSRMLS_CC );
+	object_properties_init(&intern->zo, ce);
 	intern->date_type = 0;
 	intern->time_type = 0;
 	intern->calendar  = 1;		/* Gregorian calendar */
--- php5.orig/ext/intl/formatter/formatter_class.c
+++ php5/ext/intl/formatter/formatter_class.c
@@ -62,6 +62,7 @@ zend_object_value NumberFormatter_object
 	intern = ecalloc( 1, sizeof(NumberFormatter_object) );
 	formatter_data_init( &intern->nf_data TSRMLS_CC );
 	zend_object_std_init( &intern->zo, ce TSRMLS_CC );
+	object_properties_init(&intern->zo, ce);
 
 	retval.handle = zend_objects_store_put(
 		intern,
--- php5.orig/ext/intl/resourcebundle/resourcebundle_class.c
+++ php5/ext/intl/resourcebundle/resourcebundle_class.c
@@ -63,6 +63,7 @@ static zend_object_value ResourceBundle_
 	rb = ecalloc( 1, sizeof(ResourceBundle_object) );
 
 	zend_object_std_init( (zend_object *) rb, ce TSRMLS_CC );
+	object_properties_init((zend_object *) rb, ce);
 
 	intl_error_init( INTL_DATA_ERROR_P(rb) TSRMLS_CC );
 	rb->me = NULL;
--- php5.orig/ext/intl/spoofchecker/spoofchecker_class.c
+++ php5/ext/intl/spoofchecker/spoofchecker_class.c
@@ -61,6 +61,7 @@ zend_object_value Spoofchecker_object_cr
 	intern = ecalloc(1, sizeof(Spoofchecker_object));
 	intl_error_init(SPOOFCHECKER_ERROR_P(intern) TSRMLS_CC);
 	zend_object_std_init(&intern->zo, ce TSRMLS_CC);
+	object_properties_init(&intern->zo, ce);
 
 	retval.handle = zend_objects_store_put(
 		intern,
--- php5.orig/ext/standard/incomplete_class.c
+++ php5/ext/standard/incomplete_class.c
@@ -109,6 +109,8 @@ static zend_object_value php_create_inco
 	value = zend_objects_new(&object, class_type TSRMLS_CC);
 	value.handlers = &php_incomplete_object_handlers;
 	
+	object_properties_init(object, class_type);
+	
 	return value;
 }
 
